### ISSUE TITLE
Experiment API

### ISSUE THREAD

**Issue by michelole** [bst-mug/trec2017#53]
It would be nice to have an easy way to run lots of experiments (maybe in parallel?) and get the relevant metrics in a spreadsheet.

**Comment by michelole** [comment]
@plopezgarcia suggested the following in a email:

> As I suggested a few weeks ago, at some point the code should freeze, and every experiment should have input parameters that should come from a file stored in the repository for reference and tracking, which represents a specific experiment, like:
> 
> experiment001.json:
> ---------------------------
> "{geneExpansion:true,
>   addTreatmentKeywords:true,
>   addSexDemographics:true,
>   translateDemographicsToAgeGroup:true,
> 
>   ....}"
> 
> or something of that sort.
> 
> This will enable us to run all sorts of experiments and variations in one go, analyze, and plot the results as desired and which whatever tools we want, and finally pick the experiments that provide the best results.?
> 
> Ideally, we should have two input experiment folders, one for the automatic runs, and one for the manual runs, which can even be directly the elastic search query if Stefan/Zdenko are willing to curate them really accurately.

**Comment by michelole** [comment]
TODO/Wishlist:
- addResults() deprecated
- read from JSON
- parallel processing / thread-safe
- hash Experiment and store all results with unique ID
- CSVWriter for experiments
- expose TrecEval @ Experiment class
- ExperimentBuilder should provide a single experiment
- ExperimentSetBuilder provides related experiments
- validation rules
